src/NexusMods.CrossPlatform/ProtocolRegistration/ProtocolRegistrationLinux.cs

```diff
@@ -13,124 +13,18 @@ namespace NexusMods.CrossPlatform.ProtocolRegistration;
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, "Exception while creating desktop file, the handler for `{Scheme}` might not work", uriScheme);
-                return;
-            }
-
-            try
-            {
-                await UpdateMIMECacheDatabase(applicationsDirectory, cancellationToken: cancellationToken);
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, "Exception while updating MIME cache database, see process logs for more details");
-                return;
-            }
-        }
-
-        if (setAsDefaultHandler)
-        {
-            try
-            {
-                await SetAsDefaultHandler(uriScheme, cancellationToken: cancellationToken);
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, "Exception while setting the default handler for `{Scheme}`, see the process logs for more details", uriScheme);
-            }
-        }
-    }
-
-    private async Task CreateDesktopFile(AbsolutePath applicationsDirectory, CancellationToken cancellationToken = default)
-    {
-        if (!applicationsDirectory.DirectoryExists())
-        {
-            applicationsDirectory.CreateDirectory();
-        }
-
-        var filePath = applicationsDirectory.Combine(DesktopFile);
-        var backupPath = filePath.AppendExtension(new Extension(".bak"));
-
-        if (filePath.FileExists)
-        {
-            _logger.LogInformation("Moving existing desktop file from `{From}` to `{To}`", filePath, backupPath);
-        }
-
-        _logger.LogInformation("Creating desktop file at `{Path}`", filePath);
-
-        await using var stream = typeof(ProtocolRegistrationLinux).Assembly.GetManifestResourceStream(DesktopFileResourceName);
-        if (stream is null)
-        {
-            _logger.LogError($"Manifest resource Stream for `{DesktopFileResourceName}` is null!");
-            return;
-        }
-
-        using var sr = new StreamReader(stream, encoding: Encoding.UTF8);
-
-        var text = await sr.ReadToEndAsync(cancellationToken);
-        text = text.Replace(ExecutablePathPlaceholder, EscapeWhitespaceForCli(_osInterop.GetOwnExe()));
-
-        await filePath.WriteAllTextAsync(text, cancellationToken);
     }
 
-    private async Task UpdateMIMECacheDatabase(AbsolutePath applicationsDirectory, CancellationToken cancellationToken = default)
+    public Task RegisterHandler(string uriScheme, bool setAsDefaultHandler = true, CancellationToken cancellationToken = default)
     {
-        _logger.LogInformation("Updating MIME cache database");
-
-        var command = _updateDesktopDatabaseDependency.BuildUpdateCommand(EscapeWhitespaceForCli(applicationsDirectory));
-        await _processFactory.ExecuteAsync(command, cancellationToken: cancellationToken);
+        return Task.CompletedTask;
```

Comment:
So none of the linux install methods will have protocol registration? They will all rely on package managers/custom user setup?

Reply:
Yup.